tests/virt_autotest/sev_snp_guest_verification.pm

```diff
@@ -0,0 +1,1270 @@
+
+    # Check SEV-SNP capability
+    $self->check_sev_snp_capability();
+
+    return $self;
+}
+
+=head2 activate_coco_module
+
+  activate_coco_module($self)
+
+Activates the Confidential Computing module on SLES 15-SP6 or 15-SP7 systems.
+This module is required for SEV-SNP support on these versions of SLES.
+The function determines the correct module path based on the SP version
+and uses suseconnect to activate the module.
+
+=cut
+
+sub activate_coco_module {
+    my $self = shift;
+
+    # Only needed for SLES 15-SP6 or 15-SP7
+    if (is_sle('=15-sp6') || is_sle('=15-sp7')) {
+        # Get the SP version to form the correct module path
+        my $sp_version = is_sle('=15-sp6') ? '15.6' : '15.7';
+        record_info('Activating COCO module', "Activating Confidential Computing module for SLES $sp_version");
+
+        my $module_path = "sle-module-confidential-computing/$sp_version/x86_64";
+        my $ret = script_run("suseconnect -p $module_path");
+        if ($ret != 0) {
+            record_info('Module activation failed', 'Failed to activate Confidential Computing module. This is required for SEV-SNP support.', result => 'fail');
+            die "SEV-SNP verification requires Confidential Computing module. Test cannot continue without this module activated.";
+        } else {
+            record_info('Module activated', "Successfully activated Confidential Computing module for SLES $sp_version");
+        }
+    }
+
+    return $self;
+}
+
+=head2 check_sev_snp_capability
+
+  check_sev_snp_capability($self)
+
+Check SEV-SNP capability on the host using the snphost tool ("snphost ok").
+For both SLE 15-SP6/SP7 and SLE 16+, we use the snphost tool to verify
+hardware and firmware support for SEV-SNP.
+
+This function performs a basic check to verify there are no FAIL entries 
+in the output, which would indicate missing SEV-SNP support.
+
+=cut
+
+sub check_sev_snp_capability {
+    my $self = shift;
+
+    record_info('SNP Host Check', "Running SEV-SNP capability checks to verify hardware/firmware support");
+
+    # Use snphost for both SLE 15-SP6/SP7 and SLE 16+
+    my $tool_name = "snphost";
+    my $check_cmd = "snphost ok";
+
+    # First check if the tool is available
+    my $has_tool = script_run("which $tool_name") == 0;
+
+    if (!$has_tool) {
+        record_info('SEV-SNP Tool Missing', "$tool_name tool not found, required for SEV-SNP hardware verification", result => 'fail');
+        die "$tool_name tool not found. This tool is required for SEV-SNP hardware verification.";
+    }
+
+    # Run the check command and capture the output
+    my $check_output = script_output("$check_cmd 2>&1", proceed_on_failure => 1);
+    save_screenshot;
+
+    # Just check if there are any FAIL entries in the output
+    if ($check_output =~ /FAIL/) {
+        record_info('SNP Host Check Failed', "$tool_name reports FAIL for some SEV-SNP requirements. Check the detailed output for more information.", result => 'fail');
```

Comment:
If the $check_output is more than one screen long, would be good to output it also in this record_info msg.